v2/pkg/buildassets/buildassets.go

```diff
@@ -1,67 +1,145 @@
+		// The file does not exist, let's read it from the assets FS and write it to disk
+		content, err := iofs.ReadFile(assets, file)
+		if err != nil {
+			return nil, err
+		}
+
+		if err := writeFileSystemFile(projectData, file, content); err != nil {
+			return nil, fmt.Errorf("Unable to create file in build folder: %s", err)
+		}
+		return content, nil
 	}
 
-	return nil
+	return content, err
 }
 
-func RegenerateManifest(target string) error {
-	a, err := debme.FS(assets, "build")
+// ReadFileWithProjectData reads the file from the project build folder and replaces ProjectInfo if necessary.
+// If the file does not exist it falls back to the embedded file and the file will be written
+// to the disk for customisation. The file written is the original unresolved one.
+func ReadFileWithProjectData(projectData *project.Project, file string) ([]byte, error) {
+	content, err := ReadFile(projectData, file)
 	if err != nil {
-		return err
+		return nil, err
 	}
-	return a.CopyFile("windows/wails.exe.manifest", target, 0644)
-}
 
-func RegenerateAppIcon(target string) error {
-	a, err := debme.FS(assets, "build")
+	content, err = resolveProjectData(content, projectData)
 	if err != nil {
-		return err
+		return nil, fmt.Errorf("Unable to resolve data in %s: %w", file, err)
 	}
-	return a.CopyFile("appicon.png", target, 0644)
+	return content, nil
 }
 
-func RegeneratePlist(targetDir string, projectName string) error {
-	darwinAssets, err := debme.FS(assets, "build/darwin")
+// ReadOriginalFileWithProjectDataAndSave reads the file from the embedded assets and replaces
+// ProjectInfo if necessary.
+// It will also write the resolved final file back to the project build folder.
+func ReadOriginalFileWithProjectDataAndSave(projectData *project.Project, file string) ([]byte, error) {
+	file = path.Join(rootFolder, file)
+	content, err := iofs.ReadFile(assets, file)
 	if err != nil {
-		return err
+		return nil, fmt.Errorf("Unable to read file %s: %w", file, err)
 	}
-	templateDir := gosod.New(darwinAssets)
-	err = templateDir.Extract(targetDir, &assetData{Name: projectName})
+
+	content, err = resolveProjectData(content, projectData)
 	if err != nil {
-		return err
+		return nil, fmt.Errorf("Unable to resolve data in %s: %w", file, err)
 	}
 
+	if err := writeFileSystemFile(projectData, file, content); err != nil {
+		return nil, fmt.Errorf("Unable to create file in build folder: %w", err)
+	}
+	return content, nil
+}
+
+type assetData struct {
+	Name string
+	Info project.Info
+
+	Internal struct {
+		Webview2RuntimeSetupPath string
+	}
+}
+
+func resolveProjectData(content []byte, projectData *project.Project) ([]byte, error) {
+	tmpl, err := template.New("").Parse(string(content))
+	if err != nil {
+		return nil, err
+	}
+
+	data := &assetData{
+		Name: projectData.Name,
+		Info: projectData.Info,
+	}
+
+	data.Internal.Webview2RuntimeSetupPath = filepath.Join(fs.RelativePath("../../"), "internal", "webview2runtime", "MicrosoftEdgeWebview2Setup.exe")
```

Comment:
We're moving away from assuming the Wails CLI is installed as a Go module so we don't want to reference files relative to the source. I think we should copy this setup file into a temp directory. I've added a [method](https://github.com/wailsapp/wails/blob/master/v2/internal/webview2runtime/webview2runtime.go#L175) to the webview2runtime package to do this. Perhaps we could use `os.TempDir()`?

Reply:
Yeah, I thought that initially but then it'd be easy to check in with your project. The reason I suggested temp dir was because I assumed it would be deleted anyway.

Reply:
Given everything you've said, I'd suggest dumping it in the build dir and add a .gitignore is fine 👍

Reply:
Great, I'll update the PR on monday

Reply:
.gitignore entries have been added.